Remember previous item key to invoke a trigger as the last status changed item

takyoshi opened the pull request:
Hi,

This pr is to change `lastStatus` map to store one previous item when to use trigger option.

I expected that when an item's status (Info, Warning, or Danger) changes the item's data would be input in stdin, but if category has multiple keys, a first item which is the same status as the actually changed item is picked up.

POST with curl:

``` bash
## store keys into category
$ curl -X PUT -d "Success" http://localhost:8500/v1/kv/dashboard/debug/tkyshm/key1\?flags=1422597159000
$ curl -X PUT -d "Success" http://localhost:8500/v1/kv/dashboard/debug/tkyshm/key2\?flags=1422597159000
$ curl -X PUT -d "Success" http://localhost:8500/v1/kv/dashboard/debug/tkyshm/key3\?flags=1422597159000

## change key3 status success -> danger, dagner -> success
$ curl -X PUT -d "Danger" http://localhost:8500/v1/kv/dashboard/debug/tkyshm/key3\?flags=1422597159002
$ curl -X PUT -d "key3 is successful" http://localhost:8500/v1/kv/dashboard/debug/tkyshm/key3\?flags=1422597159000
```

trigger stdin:

```
danger:  {"category":"debug","node":"tkyshm","address":"127.0.0.1","timestamp":"2015-01-30 14:52:39 +0900","status":"danger","key":"key3","data":"Danger"}
success:  {"category":"debug","node":"tkyshm","address":"127.0.0.1","timestamp":"2015-01-30 14:52:39 +0900","status":"success","key":"key1","data":"Success"}
```

I would expect that danger -> success stdin is:

```
success:  {"category":"debug","node":"tkyshm","address":"127.0.0.1","timestamp":"2015-01-30 14:52:39 +0900","status":"success","key":"key3","data":"key3 is successful"}
```

So, I changed to remember one previous item as to all items.

Thanks.

takyoshi committed:
remembers previous item key before status

takyoshi committed:
- insert all current items into `cuurentItem` map.
- changes the currentItem map's key as to be
  genereted by `item.Category` and `item.Key`.

fujiwara commented:
Thanks for the PR.

This change seems to invoke a trigger for each category/key entries.
I'm sorry, but I hope a trigger will be invoked only when category's max (=worst) status is changed. Because if the trigger is fired for each entry, it is too much for notification.

takyoshi committed:
remember last item Key each category

I canceled to remember a previous status each item because when one item
changes status, the trigger fires.
The dashboard should fires the trigger when all of items in a category
status changed.

takyoshi commented:
Thanks for your review. 
I understand and fixed the previous commit that a trigger is fired only when category's is changed.

Please take another look.

takyoshi committed:
enable to trigger with item that causes category status is changed.

- remembers all item status that registered at consul KV.

fujiwara commented:
LGTM. Thank you!
